cli-plugins/manager/cobra.go

```diff
@@ -90,11 +90,18 @@ func AddPluginCommandStubs(dockerCli command.Cli, rootCmd *cobra.Command) (err e
 					cargs = append(cargs, args...)
 					cargs = append(cargs, toComplete)
 					os.Args = cargs
-					runCommand, runErr := PluginRunCommand(dockerCli, p.Name, cmd)
+					pluginRunCommand, runErr := PluginRunCommand(dockerCli, p.Name, cmd)
 					if runErr != nil {
 						return nil, cobra.ShellCompDirectiveError
 					}
-					runErr = runCommand.Run()
+					if verifiedDockerCli, ok := dockerCli.(*command.DockerCli); ok {
+						runCommand := verifiedDockerCli.InstrumentPluginCommand(pluginRunCommand)
+						runErr = runCommand.TimedRun()
```

Comment:
`InstrumentPluginCommand` returns a non-exported type, and we always execute `TimedRun()` afterwards, except for the `tryPluginRun` function, where we construct it, and then later execute `TimedRun`.

It's unfortunate that `exec.Cmd` doesn't implement an interface, otherwise we could've made that the return, and we can't just define an interface that _only_ has a `func Run() error`

Perhaps we should either make this function perform the `TimedRun`; puny-code;

```go
if cli is a pluginInstrumenter {
    cli.RunInstrumented(pluginRunCommand)
} else {
    pluginRunCommand.Run()
}
```

But that could even be a function, because effectively it's not a method of the _CLI_;


```go
func RunPluginCommandInstrumented(cli, pluginCommand)
```

Hmm.. but then we get to `startPluginCommandTimer`, which is even more generic, but missing the `baseAttrs` 🤔

Reply:
> It's unfortunate that exec.Cmd doesn't implement an interface, otherwise we could've made that the return, and we can't just define an interface that only has a func Run() error

My thoughts exactly, it's kinda annoying. 

We can surely find a better way to do this, my original goal with the first implementation was to keep the approach as close as possible to that used to instrument the cobra commands, but maybe that's not a great idea all things considered. The `dockerCli` struct / `Cli` interface situation in the CLI (since we don't always pass around just the interface or just the struct) also caused some minor headaches and ugliness that would be nice to avoid if possible.